app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -374,13 +388,16 @@ protected function processTextualSwatch(Attribute $attribute)
      * Get option id. If it not exist get it from dependency link array
      *
      * @param integer $optionId
+     *
      * @return int
      */
     protected function getAttributeOptionId($optionId)
     {
-        if (substr($optionId, 0, 6) == self::BASE_OPTION_TITLE || substr($optionId, 0, 3) == self::API_OPTION_PREFIX) {
-            $optionId = isset($this->dependencyArray[$optionId]) ? $this->dependencyArray[$optionId] : null;
+        if (strpos((string)$optionId, self::BASE_OPTION_TITLE) === 0 ||
+            strpos((string)$optionId, self::API_OPTION_PREFIX) === 0) {
```

Comment:
Not clear what is checked here from the first glance. Can I ask you to move these conditions to another private method and name it so it displays what exactly it checks.

Reply:
To be honest I'm not sure what is purpose of this check, and I'm not author of this part. I only replaced `substr` with `strpos` since it made more sense in such comparison in my opinion...

Reply:
Ok, I think we can live with this=)